File: de.carsten_lenz.liquibasegui/src/de/carsten_lenz/liquibasegui/util/FileToStringConverter.scala
package de.carsten_lenz.liquibasegui.util

import org.eclipse.core.databinding.conversion._
import java.io.File

class FileToStringConverter extends Converter(classOf[File], classOf[String]) {
    def convert(o: Object) = o match {
        case f: File => f.getAbsolutePath
        case _ => ""
    }
}

File: de.carsten_lenz.liquibasegui/src/de/carsten_lenz/liquibasegui/util/FileToStringConverter.java
package de.carsten_lenz.liquibasegui.util;

import java.io.File;

import org.eclipse.core.databinding.conversion.Converter;

public class FileToStringConverter extends Converter {
    
    public FileToStringConverter() {
        super(File.class, String.class);
    }

    @Override
    public Object convert(Object fromObject) {
        if (fromObject instanceof File) {
            return ((File)fromObject).getAbsolutePath();
        }

        return null;
    }
}
